src/app/service/todo-http.service.spec.ts

```diff
@@ -0,0 +1,167 @@
+import { TestBed } from '@angular/core/testing';
+
+import { TodoHttpService } from './todo-http.service';
+import { HttpClient } from '@angular/common/http';
+import { ToDoItem } from 'src/model/ToDoItem';
+import { defer } from 'rxjs';
+
+function asyncData<T>(data: T) {
+  return defer(() => Promise.resolve(data));
+}
+describe('TodoHttpService', () => {
+  let service: TodoHttpService;
+  let httpClientSpy: jasmine.SpyObj<HttpClient>;
+
+  beforeEach(() => {
+    httpClientSpy = jasmine.createSpyObj('HttpClient', [
+      'get',
+      'post',
+      'put',
+      'delete',
+    ]);
+    service = new TodoHttpService(httpClientSpy);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should get all todo items when call getAll', () => {
+    httpClientSpy.get.and.returnValue(
+      asyncData([
+        {
```

Comment:
You could put this data into beforeEach(), so that you do not need to create each time.